### PR TITLE
Add deploy app label to toolbar

### PR DESCRIPTION
**Pull request by nenb**
This PR adds the deploy app label as indicated in the mockups.

I spent some time trying to find a way to dynamically update the label based on whether or not the toolbar was 'narrow'. (I defined narrow as whether or not `popupOpener` was visible on the toolbar.)

Unfortunately, there appears to be a bug in how the label is set for extensions to the toolbar - it seems to be required to set the `label` value in the JSON file inside `/schema`, and the value in `index.ts` appears to be consistently ignored. See the final comment in https://discourse.jupyter.org/t/jl-4-toolbar-button-with-icon-and-text/20519/2.

I couldn't find a way to update non `properties` attributes in the JSON file (see also https://github.com/jupyterlab/extension-examples/tree/main/settings) and so I have not included any dynamic update to the label in this PR.

## Reference Issues or PRs

https://github.com/nebari-dev/jupyterlab-jhub-apps/issues/2

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [ ] Did you add new tests?

#### Text styling

- [ ] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [ ] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [ ] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If the content were to be read as plain text, it still makes sense, and no information is missing.

### CODE REVIEW

**Comment by nenb** [comment]
bot please update snapshots

File: style/base.css [hunk 1]
```diff
@@ -3,3 +3,13 @@
 
     https://jupyterlab.readthedocs.io/en/stable/developer/css.html
 */
+
+.jp-ToolbarButtonComponent[data-command='jhub-apps:deploy-app'] svg {
```
Comment (krassowski) [inline style/base.css:7]: Can you use child combinator here? Bare selector on right hand side can impact performance of the entire app. Browser match s lectors right to left traversing entire dom tree (optimizations like bloom filters aside). Adding a more specific rule allows browser to short circuit after looking just at the parent.

Comment (nenb) [reply style/base.css]: Thanks Mike, I didn't appreciate this, really great advice! 🙏

**Review by krassowski — APPROVED** [review]
Thanks @nenb!

**Pull request merged**